stages/test/test_mkfs_xfs.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/python3
+
+import os.path
+import re
+import subprocess
+from unittest import mock
+
+import pytest
+
+from osbuild import testutil
+from osbuild.testutil import has_executable
+
+STAGE_NAME = "org.osbuild.mkfs.xfs"
+
+
+# Prepare dataset containing good and bad API call parameters
+@pytest.mark.parametrize("test_data, expected_err", [
+    # Bad API parameters
+    ({}, "'uuid' is a required property"),
+    ({"uuid": 123}, "123 is not of type 'string'"),
+    ({"uuid": "text", "label": 1234}, "1234 is not of type 'string'"),
+    ({"uuid": "text", "label": "12345678901234567"}, "is too long"),
+    # Good API parameters
+    # mkfs requires valid UUID, but API will accept any string right now.
+    # Bellow will fail when API will be more restricted in the future.
+    ({"uuid": "text"}, ""),
+    ({"uuid": "text", "label": "123456789012"}, ""),
+])
+# This test validates only API calls using correct and incorrect queries
+def test_schema_validation_mkfs_xfs(stage_schema, test_data, expected_err):
+    test_input = {
+        "type": STAGE_NAME,
+        "devices": {
+            "device": {
+                "path": "some-path",
+            },
+        },
+        "options": {
+        }
+    }
+    test_input["options"].update(test_data)
+    res = stage_schema.validate(test_input)
+
+    if expected_err == "":
+        assert res.valid is True, f"err: {[e.as_dict() for e in res.errors]}"
+    else:
+        assert res.valid is False
+        testutil.assert_jsonschema_error_contains(res, expected_err, expected_num_errs=1)
+
+
+# This test creates dummy filesystem using predefined parameters and verifies
+# that end result has those parameters set
+@pytest.mark.skipif(not has_executable("mkfs.xfs"), reason="need mkfs.xfs")
+def test_mkfs_fat_integration(tmp_path, stage_module):
+    fake_disk_path = tmp_path / "fake.img"
+    with fake_disk_path.open("w"):
```

Comment:
Fixed:)

Reply:
I'm sorry, what I wrote above is misleading (and partly wrong because os.truncate() needs an existing file). But I think combining the two is slightly nicer:
```
diff --git a/stages/test/test_mkfs_xfs.py b/stages/test/test_mkfs_xfs.py
index 0cfcd7c3..cdf84baf 100644
--- a/stages/test/test_mkfs_xfs.py
+++ b/stages/test/test_mkfs_xfs.py
@@ -54,8 +54,8 @@ def test_schema_validation_mkfs_xfs(stage_schema, test_data, expected_err):
 @pytest.mark.skipif(not has_executable("mkfs.xfs"), reason="need mkfs.xfs")
 def test_mkfs_xfs_integration(tmp_path, stage_module):
     fake_disk_path = tmp_path / "fake.img"
-    with fake_disk_path.open("w"):
-        os.truncate(os.fspath(fake_disk_path), 300 * 1024 * 1024)
+    with fake_disk_path.open("w") as fp:
+        fp.truncate(300 * 1024 * 1024)
     devices = {
         "device": {
             "path": fake_disk_path,
```
this should work and it feels more natural IMHO as we open the file descriptor and then use that fd to truncate.

Alternatively:
```
diff --git a/stages/test/test_mkfs_xfs.py b/stages/test/test_mkfs_xfs.py
index 0cfcd7c3..dddc01b8 100644
--- a/stages/test/test_mkfs_xfs.py
+++ b/stages/test/test_mkfs_xfs.py
@@ -54,8 +54,8 @@ def test_schema_validation_mkfs_xfs(stage_schema, test_data, expected_err):
 @pytest.mark.skipif(not has_executable("mkfs.xfs"), reason="need mkfs.xfs")
 def test_mkfs_xfs_integration(tmp_path, stage_module):
     fake_disk_path = tmp_path / "fake.img"
-    with fake_disk_path.open("w"):
-        os.truncate(os.fspath(fake_disk_path), 300 * 1024 * 1024)
+    fake_disk_path.touch()
+    os.truncate(fake_disk_path, 300 * 1024 * 1024)
     devices = {
         "device": {
             "path": fake_disk_path,
```
:)

Reply:
Thanks for the explanation, I have updated all `mkfs` related PRs with this change:)